### ISSUE TITLE
http: panic serving 127.0.0.1:49439: runtime error: slice bounds out of range

### ISSUE THREAD

**Issue by attilagyorffy** [puma/puma-dev#34]
When running puma-dev in the foreground (as i couldn't get it to work in the background) I am finding some strange output that references @evanphx's own path :) Here's my terminal output:

```
$ puma-dev
* Directory for apps: /Users/attila/.puma-dev
* Domains: dev
* DNS Server port: 9253
* HTTP Server port: 9280
* HTTPS Server port: 9283
! Puma dev listening on http and https
2016/08/03 17:52:07 http: panic serving 127.0.0.1:49439: runtime error: slice bounds out of rangegoroutine 4 [running]:
net/http.(*conn).serve.func1(0xc82015a000)
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:1389 +0xc1
panic(0x4438e00, 0xc82000a090)
	/usr/local/Cellar/go/1.6.3/libexec/src/runtime/panic.go:443 +0x4e9
puma/dev.(*HTTPServer).director(0xc820121e90, 0xc8201600e0, 0x0, 0x0)
	/Users/evan/git/puma-dev/src/puma/dev/http.go:69 +0x193
puma/dev.(*HTTPServer).(puma/dev.director)-fm(0xc8201600e0, 0x0, 0x0)
	/Users/evan/git/puma-dev/src/puma/dev/http.go:79 +0x38
puma/httputil.(*ReverseProxy).ServeHTTP(0xc820124eb0, 0x4e401f8, 0xc8201500d0, 0xc820160000)
	/Users/evan/git/puma-dev/src/puma/httputil/reverseproxy.go:182 +0x4cb
net/http.serverHandler.ServeHTTP(0xc820116f00, 0x4e401f8, 0xc8201500d0, 0xc820160000)
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:2081 +0x19e
net/http.(*conn).serve(0xc82015a000)
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:1472 +0xf2e
created by net/http.(*Server).Serve
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:2137 +0x44e
2016/08/03 17:52:07 http: panic serving 127.0.0.1:49441: runtime error: slice bounds out of range
goroutine 6 [running]:
net/http.(*conn).serve.func1(0xc82015a280)
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:1389 +0xc1
panic(0x4438e00, 0xc82000a090)
	/usr/local/Cellar/go/1.6.3/libexec/src/runtime/panic.go:443 +0x4e9
puma/dev.(*HTTPServer).director(0xc820121e90, 0xc82012a0e0, 0x0, 0x0)
	/Users/evan/git/puma-dev/src/puma/dev/http.go:69 +0x193
puma/dev.(*HTTPServer).(puma/dev.director)-fm(0xc82012a0e0, 0x0, 0x0)
	/Users/evan/git/puma-dev/src/puma/dev/http.go:79 +0x38
puma/httputil.(*ReverseProxy).ServeHTTP(0xc820124eb0, 0x4e401f8, 0xc8200e2c30, 0xc8201601c0)
	/Users/evan/git/puma-dev/src/puma/httputil/reverseproxy.go:182 +0x4cb
net/http.serverHandler.ServeHTTP(0xc820116f00, 0x4e401f8, 0xc8200e2c30, 0xc8201601c0)
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:2081 +0x19e
net/http.(*conn).serve(0xc82015a280)
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:1472 +0xf2e
created by net/http.(*Server).Serve
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:2137 +0x44e
2016/08/03 17:52:07 http: panic serving 127.0.0.1:49443: runtime error: slice bounds out of range
goroutine 7 [running]:
net/http.(*conn).serve.func1(0xc82015a300)
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:1389 +0xc1
panic(0x4438e00, 0xc82000a090)
	/usr/local/Cellar/go/1.6.3/libexec/src/runtime/panic.go:443 +0x4e9
puma/dev.(*HTTPServer).director(0xc820121e90, 0xc820160380, 0x0, 0x0)
	/Users/evan/git/puma-dev/src/puma/dev/http.go:69 +0x193
puma/dev.(*HTTPServer).(puma/dev.director)-fm(0xc820160380, 0x0, 0x0)
	/Users/evan/git/puma-dev/src/puma/dev/http.go:79 +0x38
puma/httputil.(*ReverseProxy).ServeHTTP(0xc820124eb0, 0x4e401f8, 0xc820150270, 0xc8201602a0)
	/Users/evan/git/puma-dev/src/puma/httputil/reverseproxy.go:182 +0x4cb
net/http.serverHandler.ServeHTTP(0xc820116f00, 0x4e401f8, 0xc820150270, 0xc8201602a0)
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:2081 +0x19e
net/http.(*conn).serve(0xc82015a300)
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:1472 +0xf2e
created by net/http.(*Server).Serve
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:2137 +0x44e
2016/08/03 17:52:11 http: panic serving 127.0.0.1:49446: runtime error: slice bounds out of range
goroutine 26 [running]:
net/http.(*conn).serve.func1(0xc820117180)
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:1389 +0xc1
panic(0x4438e00, 0xc82000a090)
	/usr/local/Cellar/go/1.6.3/libexec/src/runtime/panic.go:443 +0x4e9
puma/dev.(*HTTPServer).director(0xc820121e90, 0xc820160460, 0x0, 0x0)
	/Users/evan/git/puma-dev/src/puma/dev/http.go:69 +0x193
puma/dev.(*HTTPServer).(puma/dev.director)-fm(0xc820160460, 0x0, 0x0)
	/Users/evan/git/puma-dev/src/puma/dev/http.go:79 +0x38
puma/httputil.(*ReverseProxy).ServeHTTP(0xc820124eb0, 0x4e401f8, 0xc820150410, 0xc82012a1c0)
	/Users/evan/git/puma-dev/src/puma/httputil/reverseproxy.go:182 +0x4cb
net/http.serverHandler.ServeHTTP(0xc820116f00, 0x4e401f8, 0xc820150410, 0xc82012a1c0)
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:2081 +0x19e
net/http.(*conn).serve(0xc820117180)
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:1472 +0xf2e
created by net/http.(*Server).Serve
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:2137 +0x44e
2016/08/03 17:52:11 http: panic serving 127.0.0.1:49448: runtime error: slice bounds out of range
goroutine 36 [running]:
net/http.(*conn).serve.func1(0xc8201b4000)
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:1389 +0xc1
panic(0x4438e00, 0xc82000a090)
	/usr/local/Cellar/go/1.6.3/libexec/src/runtime/panic.go:443 +0x4e9
puma/dev.(*HTTPServer).director(0xc820121e90, 0xc8201ba0e0, 0x0, 0x0)
	/Users/evan/git/puma-dev/src/puma/dev/http.go:69 +0x193
puma/dev.(*HTTPServer).(puma/dev.director)-fm(0xc8201ba0e0, 0x0, 0x0)
	/Users/evan/git/puma-dev/src/puma/dev/http.go:79 +0x38
puma/httputil.(*ReverseProxy).ServeHTTP(0xc820124eb0, 0x4e401f8, 0xc82013c0d0, 0xc8201ba000)
	/Users/evan/git/puma-dev/src/puma/httputil/reverseproxy.go:182 +0x4cb
net/http.serverHandler.ServeHTTP(0xc820116f00, 0x4e401f8, 0xc82013c0d0, 0xc8201ba000)
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:2081 +0x19e
net/http.(*conn).serve(0xc8201b4000)
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:1472 +0xf2e
created by net/http.(*Server).Serve
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:2137 +0x44e
2016/08/03 17:52:11 http: panic serving 127.0.0.1:49450: runtime error: slice bounds out of range
goroutine 38 [running]:
net/http.(*conn).serve.func1(0xc8201b4200)
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:1389 +0xc1
panic(0x4438e00, 0xc82000a090)
	/usr/local/Cellar/go/1.6.3/libexec/src/runtime/panic.go:443 +0x4e9
puma/dev.(*HTTPServer).director(0xc820121e90, 0xc8201ba2a0, 0x0, 0x0)
	/Users/evan/git/puma-dev/src/puma/dev/http.go:69 +0x193
puma/dev.(*HTTPServer).(puma/dev.director)-fm(0xc8201ba2a0, 0x0, 0x0)
	/Users/evan/git/puma-dev/src/puma/dev/http.go:79 +0x38
puma/httputil.(*ReverseProxy).ServeHTTP(0xc820124eb0, 0x4e401f8, 0xc82013c270, 0xc8201ba1c0)
	/Users/evan/git/puma-dev/src/puma/httputil/reverseproxy.go:182 +0x4cb
net/http.serverHandler.ServeHTTP(0xc820116f00, 0x4e401f8, 0xc82013c270, 0xc8201ba1c0)
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:2081 +0x19e
net/http.(*conn).serve(0xc8201b4200)
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:1472 +0xf2e
created by net/http.(*Server).Serve
	/usr/local/Cellar/go/1.6.3/libexec/src/net/http/server.go:2137 +0x44e
```

**Comment by attilagyorffy** [comment]
Probably worth mentioning that I installed puma-dev from the brew formula. I wonder whether that contains precompiled binaries that reference the wrong path?

**Comment by attilagyorffy** [comment]
So I took the time to install a go development env on my machine (first time here), pulled down the code and compiled my own binary and used that to follow the install instructions directly instead of using the binary that the brew formula gives.

Now I managed to run an example brand new Rails 5 app (yay!) but seems like there's still something missing from the fish shell compatibility. Please see the logs I published in [this gist](https://gist.github.com/attilagyorffy/bdc7e4efde2e8e23da72aee8cfc0df94).

Evan, is there anything I could try to help you with to isolate these issues that may be caused by fish's non-POSIX layer?

Thank you for this awesome piece of code, I'm really happy someone took the effort to push things forward. <3

**Comment by evanphx** [comment]
I'll have a release out tomorrow that will fix the fish shell issues (should also fix the go panic issue as well).

**Comment by evanphx** [comment]
This is fixed in b889de72
